.github/workflows/release.yml

```diff
@@ -0,0 +1,144 @@
+name: Release
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  release-check:
+    name: Check if version changed
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: main
+
+      - name: Use Node.js from nvmrc
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: 'js/.nvmrc'
+
+      - name: Check if version has been updated
+        id: check
+        working-directory: js
+        uses: EndBug/version-check@v2
+          
+    outputs:
+      publish: ${{ steps.check.outputs.changed }}
+
+  release:
+    name: Release
+    needs: release-check
+    if: ${{ needs.release-check.outputs.publish == 'true' }}
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: main
+
+      - name: Use Node.js from nvmrc
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: 'js/.nvmrc'
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Get version
+        id: package-version
+        working-directory: js
```

Comment:
Are you sure about this?

Reply:
Nevermind, I see that other steps are using it.

Reply:
I could be wrong but since all the JS including the `package.json` is in `js/` I thought this would be necessary.